### PR TITLE
[DEVX-469]: Patch Operations Addition for Input Annotations & Concepts

### PR DESCRIPTION
**Pull request by srikanthbachala20**
### What
- Patch Operations Addition for Input Annotations & Concepts
- Input proto bug fix

### Why

* To patch the existing input annotations & concepts
* Ref : https://clarifai.atlassian.net/browse/DEVX-469
* To avoid errors avail patching inputs

### How

* Create a function each for patching input annotations & concepts using Clarifai Grpc library
* Changed the dataset_id argument of _get_proto function from positional argument to optional argument for ease of usage

### CODE REVIEW

File: clarifai/client/input.py [hunk 1]
```diff
@@ -499,38 +504,64 @@ def get_bbox_proto(input_id: str, label: str, bbox: List, label_id: str = None)
...
-                )),
-                data=resources_pb2.Data(concepts=[
-                    resources_pb2.Concept(
-                        id=f"id-{''.join(label.split(' '))}", name=label, value=1.)
-                    if not label_id else resources_pb2.Concept(id=label_id, name=label, value=1.)
-                ]))
-        ]))
+    if annot_id:
+      input_annot_proto = resources_pb2.Annotation(
+          id=annot_id,
```
Comment (sanjaychelliah) [inline clarifai/client/input.py:509]: Is `annot_id` being None giving error? If thats the case, we can assign `resouce_pb2.Data` in a temp variable and reuse it in the if else condition.

File: clarifai/client/input.py [hunk 2]
```diff
@@ -541,23 +572,43 @@ def get_mask_proto(input_id: str, label: str, polygons: List[List[float]],
...
-                    ])),
-                data=resources_pb2.Data(concepts=[
-                    resources_pb2.Concept(
-                        id=f"id-{''.join(label.split(' '))}", name=label, value=1.)
-                    if not label_id else resources_pb2.Concept(id=label_id, name=label, value=1.)
-                ]))
-        ]))
+    if annot_id:
+      input_mask_proto = resources_pb2.Annotation(
+          id=annot_id,
```
Comment (sanjaychelliah) [inline clarifai/client/input.py:577]: Same as the `get_bbox_proto` function comment!

File: clarifai/client/input.py [hunk 3]
```diff
@@ -761,6 +811,65 @@ def upload_annotations(self, batch_annot: List[resources_pb2.Annotation], show_l
 
     return retry_upload
 
+  def patch_annotations(self, batch_annot: List[resources_pb2.Annotation],
+                        action: str = 'merge') -> None:
+    """Upload image annotations to app.
```
Comment (sanjaychelliah) [inline clarifai/client/input.py:816]: Description of the function is wrong here.

File: clarifai/client/input.py [hunk 4]
```diff
@@ -761,6 +811,65 @@ def upload_annotations(self, batch_annot: List[resources_pb2.Annotation], show_l
...
+        self.logger.warning(f"Patch annotations failed due to {response.status}")
+    else:
+      self.logger.info("\nPatch Annotations Uploaded Successful\n%s", response.status)
+
+  def patch_concepts(self,
+                     concept_ids: List[str],
+                     labels: List[str],
+                     values: List[float] = [],
+                     action: str = 'overwrite') -> None:
+    """Upload image annotations to app.
```
Comment (sanjaychelliah) [inline clarifai/client/input.py:844]: Description of the function is wrong here too

File: clarifai/client/input.py [hunk 5]
```diff
@@ -761,6 +811,65 @@ def upload_annotations(self, batch_annot: List[resources_pb2.Annotation], show_l
...
+          if annot['status']['code'] != status_code_pb2.ANNOTATION_SUCCESS:
+            self.logger.warning(f"Patch annotations failed, status: {annot['status']}")
+      except Exception:
+        self.logger.warning(f"Patch annotations failed due to {response.status}")
+    else:
+      self.logger.info("\nPatch Annotations Uploaded Successful\n%s", response.status)
+
+  def patch_concepts(self,
+                     concept_ids: List[str],
+                     labels: List[str],
```
Comment (sanjaychelliah) [inline clarifai/client/input.py:841]: `labels` param can be None, If not provided we can reuse the `concept_ids` as `labels`

File: clarifai/client/input.py (hunk 3 above)
Comment (srikanthbachala20) [reply clarifai/client/input.py:816]: Done

File: clarifai/client/input.py (hunk 1 above)
Comment (srikanthbachala20) [reply clarifai/client/input.py:509]: Done

File: clarifai/client/input.py (hunk 2 above)
Comment (srikanthbachala20) [reply clarifai/client/input.py:577]: Done

File: clarifai/client/input.py (hunk 5 above)
Comment (srikanthbachala20) [reply clarifai/client/input.py:841]: Done

**Review by sanjaychelliah — APPROVED** [review]
LGTM!

**Pull request merged**